contentctl/objects/abstract_security_content_objects/detection_abstract.py

```diff
@@ -123,98 +141,97 @@ def validate_test_groups(cls, value:Union[None, List[TestGroup]], info:Validatio
 
         # iterate over the unit tests and create a TestGroup (and as a result, an IntegrationTest) for each
         test_groups: list[TestGroup] = []
-        for unit_test in info.data.get("tests"):
-            test_group = TestGroup.derive_from_unit_test(unit_test, info.data.get("name"))
+        tests: list[UnitTest | IntegrationTest] = info.data.get("tests")                            # type: ignore
+        unit_test: UnitTest
+        for unit_test in tests:                                                                     # type: ignore
+            test_group = TestGroup.derive_from_unit_test(unit_test, info.data.get("name"))          # type: ignore
             test_groups.append(test_group)
 
         # now add each integration test to the list of tests
         for test_group in test_groups:
-            info.data.get("tests").append(test_group.integration_test)
+            tests.append(test_group.integration_test)
         return test_groups
 
-
     @computed_field
     @property
-    def datamodel(self)->List[DataModel]:
+    def datamodel(self) -> List[DataModel]:
         if isinstance(self.search, str):
             return [dm for dm in DataModel if dm.value in self.search]
         else:
             return []
-    
 
     @computed_field
     @property
-    def source(self)->str:
+    def source(self) -> str:
+        # TODO (cmcginley): if filepath cannot be None for a detection, we should be enforcing that
+        #   with a validator and a default value/factory
         if self.file_path is not None:
```

Comment:
We have `file_path` as a required field; this check is irrelevant no? Pydantic validation precludesthe need for this check

Reply:
Agreed, I think was a relic from a time when file_path was not a required_field (and that functionality doesn't even work right now, since it raises an exception).  I have removed that check and now just returned the parent name.